extensions/terminal-suggest/src/completions/code.ts

```diff
@@ -316,3 +331,38 @@ const codeCompletionSpec: Fig.Spec = {
 };
 
 export default codeCompletionSpec;
+
+async function commandExists(command: string): Promise<boolean> {
+	return new Promise(resolve => {
+		const paths = (process.env.PATH || '').split(path.delimiter);
+		for (const p of paths) {
+			const fullPath = path.join(p, command);
+			fs.access(fullPath, fs.constants.X_OK, err => {
+				if (!err) {
+					resolve(true);
+				}
+			});
+		}
+		resolve(false);
+	});
+}
+
+export async function getInstalledExtensions(): Promise<Fig.Suggestion[] | undefined> {
+	return new Promise((resolve, reject) => {
+		commandExists('code').then(() => {
+			spawnHelper('code', ['--list-extensions', '--show-versions'], { stdio: 'pipe', env: { ...process.env, PATH: process.env.PATH } }).then(stdout => {
```

Comment:
I think we'd want a script helper or to use from the custom callback here since we can't upstream `spawnHelper` or `commandExists`. We can't touch node in these unless it's a customization we're not planning on upstreaming.